src/api/cpp/nixl_v2_2.h

```diff
@@ -0,0 +1,125 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved. SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef _NIXL_H
+#define _NIXL_H
+
+#include <cstdint>
+#include <map>
+#include <string>
+#include <vector>
+
+namespace nixl {
+
+class Config {
+};
+
+class MemDesc {
+public:
+    enum class MemType { DRAM, VRAM, BLK, OBJ, FILE, PODMEM };
+
+    // Local memory
+    MemDesc(uint64_t address, size_t length, MemType type, uint64_t dev_id);
+
+    // Remote memory
+    MemDesc(uint64_t address, size_t length, MemType type, uint64_t dev_id,
+            std::string_view agent_name);
+};
+
+class Agent {
+public:
+    class XferList;
+    class XferRequest;
+    class OptionalParams;
+
+    Agent(std::string_view name, const Config &cfg);
```

Comment:
Wouldn't an agent instance need to save its name somewhere? If yes, why this is not a const string &?

Reply:
I mean, one could argue that in cases where extra copying is a concern it would be better to just make the argument std::string so that it would be copied once during the call to the constructor and then used to move-construct m_name instead of always having intermediate string_view object. But this gets rather theoretical considering we are talking about a class that would realistically get instantiated once during the application startup so I'm not going to argue about it further.